lib/lints/use_common_design_system_widget/node_handler/widget_node_handler.dart

```diff
@@ -0,0 +1,91 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:chili_custom_lints/lints/use_common_design_system_widget/widget_helper/widget_helper.dart';
+import 'package:collection/collection.dart';
+
+typedef DirectionalSpacing = (double? horizontal, double? vertical);
+
+class WidgetNodeHandler {
+  static String getCorrectionName(
+    String prefix,
+    double? value,
+    DesignSystemElement element,
+  ) =>
+      '$prefix${element.name}${value?.toInt()}';
+
+  static String getCorrectionMessage(String correctionName) =>
+      "Use common design system widget '$correctionName' instead";
+
+  static bool hasLintError(
+    double? horizontal,
+    double? vertical,
+    NodeList<Expression> arguments,
```

Comment:
I've just noticed this, but do you actually need to pass the whole object every time if you only check its emptiness? Could also do an early return for this check.